### PR TITLE
Adds the possibility to create dependencies between two SingleSelects

### PR DESCRIPTION
**Pull request by bkositza**
If the selection options of a SingleSelect are dependent on the selection of another SingleSelect, this can now be specified via a dependencyField and the dependent fields can be determined via autocomplete.

image

Whenever we select a classification system, only the classes that belong to it will be specified as an option.

### CODE REVIEW

File: src/main/resources/default/taglib/t/singleSelect.html.pasta [hunk 1]
```diff
@@ -12,6 +12,7 @@
 <i:arg name="optional" type="boolean" default="false"/>
 <i:arg name="allowCustomEntries" type="boolean" default="false"/>
 <i:arg name="suggestionUri" type="String" default=""/>
+<i:arg name="dependencyField" type="String" default=""/>
```
Comment (sabieber) [inline src/main/resources/default/taglib/t/singleSelect.html.pasta:15]: **Optional:** `dependencyFieldId` would be more descriptive

File: src/main/resources/default/taglib/t/singleSelect.html.pasta [hunk 2]
```diff
@@ -51,6 +52,9 @@
 </div>
 <script>
     sirius.ready(function () {
+        const _dependencyField = document.getElementById('@dependencyField');
+        const _dependencyFieldSelect = document.getElementById('@dependencyField' + '-select');
```
Comment (sabieber) [inline src/main/resources/default/taglib/t/singleSelect.html.pasta:56]: **Required:** If the other field is also a TokenAutocomplete the [val](https://github.com/scireum/token-autocomplete/blob/master/lib/token-autocomplete.ts#L288C5-L288C5) function should be used instead of relying on the internal structure of the component. This could change in the future and break this functionality.

File: src/main/resources/default/taglib/t/singleSelect.html.pasta [hunk 3]
```diff
@@ -63,8 +67,16 @@
...
-                suggestionsUri: '@suggestionUri'
+                suggestionsUri: '@suggestionUri',
+                suggestionsUriBuilder: function (query) {
+                    return this.suggestionsUri + (_dependencyFieldSelect ? "/" + _dependencyFieldSelect.value : '') + '?query=' + query;
+                }
             }, <i:render name="autocompleteOptions"/>));
+        if (_dependencyField) {
+            _dependencyField.addEventListener("tokens-changed", function() {
+                autocomplete.autocomplete.loadSuggestions();
+            });
```
Comment (sabieber) [inline src/main/resources/default/taglib/t/singleSelect.html.pasta:78]: **Question:** Why is this necessary? Suggestions should be loaded when the input field gains focus and it should not have focus when another input field changes (as in the user selects one of the suggestions of another field).

File: src/main/resources/default/taglib/t/singleSelect.html.pasta [hunk 4]
```diff
@@ -63,8 +67,16 @@
                 noMatchesText: '@i18n("template.html.selects.noMatches")',
                 noMatchesCustomEntriesDescription: '@i18n("template.html.selects.useCustomEntry")',
                 placeholderText: '@placeholder',
-                suggestionsUri: '@suggestionUri'
+                suggestionsUri: '@suggestionUri',
+                suggestionsUriBuilder: function (query) {
+                    return this.suggestionsUri + (_dependencyFieldSelect ? "/" + _dependencyFieldSelect.value : '') + '?query=' + query;
```
[the remark this answers is not in the document]
Comment (idlira) [reply src/main/resources/default/taglib/t/singleSelect.html.pasta:72]: I vote for parameter.

**Review by mko-sci — APPROVED** [review]

File: src/main/resources/default/taglib/t/singleSelect.html.pasta (hunk 3 above)
Comment (bkositza) [reply src/main/resources/default/taglib/t/singleSelect.html.pasta]: It was just a thought that the dropdown opens as soon as a value is selected in the dependent field. A kind of hint for the user, but of course you're right, we can probably leave that out.

**Review by mkeckmkeck — APPROVED** [review]

**Review by idlira — CHANGES_REQUESTED** [review]

File: src/main/resources/default/taglib/t/singleSelect.html.pasta [hunk 5]
```diff
@@ -69,14 +68,13 @@
                 placeholderText: '@placeholder',
                 suggestionsUri: '@suggestionUri',
                 suggestionsUriBuilder: function (query) {
-                    return this.suggestionsUri + (_dependencyFieldSelect ? "/" + _dependencyFieldSelect.value : '') + '?query=' + query;
+                    let baseUri = this.suggestionsUri + '?query=' + query;
+                    if (_dependencyField) {
+                        return baseUri + '&dependentValue=' + _dependencyField.tokenAutocomplete.val().join();
```
[the remark this answers is not in the document]
Comment (bkositza) [reply src/main/resources/default/taglib/t/singleSelect.html.pasta]: Good point, i encoded both now, the dependentValue and the query

**Review by idlira — APPROVED** [review]

**Review by sabieber — APPROVED** [review]
